diff --git a/configure.ac b/configure.ac
index fa62b5fa85..a563115033 100644
--- a/configure.ac
+++ b/configure.ac
@@ -443,10 +443,15 @@ AS_CASE(["$target_os"],
 		    [[FILE* volatile f = stdin; return 0;]])],
 		    [rb_cv_msvcrt=`$OBJDUMP -p conftest$ac_exeext |
 				   tr A-Z a-z |
-				   sed -n '/^[[ 	]]*dll name: \(msvc.*\)\.dll$/{s//\1/p;q;}'`],
+				   sed -n '/^[[ 	]]*dll name: \(msvc.*\)\.dll$/{s//\1/p;q;}'`
+		     test "$rb_cv_msvcrt" = "" && rb_cv_msvcrt=`$OBJDUMP -p conftest$ac_exeext |
+				   tr A-Z a-z |
+				   sed -n '/^[[ 	]]*dll name: \(api-ms-win-crt.*\)\.dll$/{s//ucrt/p;q;}'`
+		     test "$rb_cv_msvcrt" = "ucrt" && RT_VER=140
+				   ],
 		    [rb_cv_msvcrt=msvcrt])
 	test "$rb_cv_msvcrt" = "" && rb_cv_msvcrt=msvcrt])
-	RT_VER=`echo "$rb_cv_msvcrt" | tr -cd [0-9]`
+	test "$RT_VER" = "" && RT_VER=`echo "$rb_cv_msvcrt" | tr -cd [0-9]`
 	test "$RT_VER" = "" && RT_VER=60
 	AC_DEFINE_UNQUOTED(RUBY_MSVCRT_VERSION, $RT_VER)
 	sysconfdir=
diff --git a/win32/win32.c b/win32/win32.c
index 940a85a4de..b7fcd666f1 100644
--- a/win32/win32.c
+++ b/win32/win32.c
@@ -888,6 +888,12 @@ socklist_delete(SOCKET *sockp, int *flagp)
 }
 
 static int w32_cmdvector(const WCHAR *, char ***, UINT, rb_encoding *);
+
+#if RUBY_MSVCRT_VERSION >= 80
+#include <crtdbg.h>
+static void set_pioinfo_extra(void);
+#endif
+
 //
 // Initialization stuff
 //
@@ -896,11 +902,11 @@ void
 rb_w32_sysinit(int *argc, char ***argv)
 {
 #if RUBY_MSVCRT_VERSION >= 80
-    static void set_pioinfo_extra(void);
-
     _CrtSetReportMode(_CRT_ASSERT, 0);
     _set_invalid_parameter_handler(invalid_parameter);
+#if !defined(__MINGW32__)
     _RTC_SetErrorFunc(rtc_error_handler);
+#endif
     set_pioinfo_extra();
 #endif
     SetErrorMode(SEM_FAILCRITICALERRORS|SEM_NOGPFAULTERRORBOX);
