diff --git a/lib/erb.gemspec b/lib/erb.gemspec
index f58f61be3a..cc0e2c9ee6 100644
--- a/lib/erb.gemspec
+++ b/lib/erb.gemspec
@@ -16,7 +16,7 @@
   spec.files         = Dir.chdir(File.expand_path("..", __FILE__)) do
     `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
   end
-  spec.bindir        = "exe"
-  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.bindir        = "libexe"
+  spec.executables   = %w[erb]
   spec.require_paths = ["lib"]
 end
diff --git a/libexec/erb b/libexec/erb
index 2435224fe1..ded76991dc 100755
--- a/libexec/erb
+++ b/libexec/erb
@@ -171,6 +171,4 @@ EOU
   end
 end
 
-if __FILE__ == $0
-  ERB::Main.run
-end
+ERB::Main.run
