diff --git a/spec/ruby/language/magic_comment_spec.rb b/spec/ruby/language/magic_comment_spec.rb
index f2bf3a08e5..98f424c3d0 100644
--- a/spec/ruby/language/magic_comment_spec.rb
+++ b/spec/ruby/language/magic_comment_spec.rb
@@ -3,7 +3,13 @@
 # See core/kernel/eval_spec.rb for more magic comments specs for eval()
 describe :magic_comments, shared: true do
   before :each do
-    @default = @method == :locale ? Encoding.find('locale') : Encoding::UTF_8
+    @default = if @method == :locale &&
+        ( platform_is_not(:windows) || ruby_version_is(""..."4.0") )
+    then
+      Encoding.find('locale')
+    else
+      Encoding::UTF_8
+    end
   end
 
   it "are optional" do
diff --git a/test/ruby/test_rubyoptions.rb b/test/ruby/test_rubyoptions.rb
index ac4b5870eb..fac0901fdf 100644
--- a/test/ruby/test_rubyoptions.rb
+++ b/test/ruby/test_rubyoptions.rb
@@ -1082,19 +1082,11 @@ def test_command_line_glob_nonascii
 
     def test_command_line_progname_nonascii
       bug10555 = '[ruby-dev:48752] [Bug #10555]'
-      name = expected = nil
-      unless (0x80..0x10000).any? {|c|
-               name = c.chr(Encoding::UTF_8)
-               expected = name.encode("locale") rescue nil
-             }
-        omit "can't make locale name"
-      end
-      name << ".rb"
-      expected << ".rb"
+      name = "\u{20ac}.rb"
       with_tmpchdir do |dir|
         open(name, "w") {|f| f.puts "puts File.basename($0)"}
-        assert_in_out_err([name], "", [expected], [],
-                          bug10555, encoding: "locale")
+        assert_in_out_err([name], "", [name], [],
+                          bug10555, encoding: "utf-8")
       end
     end
 
-- 
2.49.0

